earth2grid/projections.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+
+import numpy as np
+import torch
+
+from earth2grid import base
+from earth2grid._regrid import BilinearInterpolator
+
+try:
+    import pyvista as pv
+except ImportError:
+    pv = None
+
+
+class Projection(abc.ABC):
+    @abc.abstractmethod
+    def project(self, lat: np.ndarray, lon: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Compute the projected x,y from lat,lon.
+        """
+        pass
+
+    @abc.abstractmethod
+    def inverse_project(self, x: np.ndarray, y: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Compute the lat,lon from the projected x,y.
+        """
+        pass
+
+
+class Grid(base.Grid):
```

Comment:
I just renamed your LCC grid class @simonbyrne . Intended to be used like `earth2grid.projections.Grid`.

Reply:
Not sure if you want to give it a more specific name (i.e. it assumes the underlying grid is rectilinear, not an unstructured mesh).

Reply:
I think "projection" implies rectangle. This is just a style preference for `projection.Grid` I have for `projection.ProjectionGrid`. Same style as `healpix.Grid`.